app/models/user.rb

```diff
@@ -46,10 +46,14 @@ def calculate_frecency
                                 .limit(num_orders)
                                 .distinct
                                 .pluck(:created_at)
-    self.frecency = last_datetimes.map(&:to_time).map(&:to_i).sum / (num_orders * 10)
+    self.frecency = (last_datetimes.map(&:to_time).map(&:to_i).sum / (num_orders * 10)) * self.multiplier
     self.save
   end
 
+  def multiplier
+    Math.atan(self.balance / 10) + (Math::PI / 2)
```

Comment:
why the Math::PI / 2?

Reply:
Makes the result of Math.atan positive.

Reply:
Why the `/10`? As atan is a monotonic function this will not change the order. (Same for the `Math::PI/2`, but it's nice to keep things positive.)